src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -52,18 +52,20 @@ def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[st
         grafana_data_sources = self.mgr.template.render(
             'services/grafana/ceph-dashboard.yml.j2', {'hosts': prom_services, 'loki_host': loki_host})
 
-        cert = self.mgr.get_store('grafana_crt')
-        pkey = self.mgr.get_store('grafana_key')
+        cert_path = f'{daemon_spec.host}/grafana_crt'
+        key_path = f'{daemon_spec.host}/grafana_key'
+        cert = self.mgr.get_store(cert_path)
+        pkey = self.mgr.get_store(key_path)
```

Comment:
LGTM, instead of the global key this change looks for keys associated to the host enrolled in the Ceph cluster.
There's a WIP PR on the OpenStack side to make sure the key/cert import is aligned.

[1] https://review.opendev.org/c/openstack/tripleo-ansible/+/854429

Reply:
> LGTM, instead of the global key this change looks for keys associated to the host enrolled in the Ceph cluster. There's a WIP PR on the OpenStack side to make sure the key/cert import is aligned.
> 
> [1] https://review.opendev.org/c/openstack/tripleo-ansible/+/854429

Awesome, then we can go ahead with the change once it passed our theuthology testing :+1: